sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -318,18 +343,17 @@ def test_sparse_encode_positivity(positive):
     rng = np.random.RandomState(0)
     V = rng.randn(n_components, n_features)  # random init
     V /= np.sum(V ** 2, axis=1)[:, np.newaxis]
-    for algo in ('lasso_lars', 'lasso_cd', 'lars', 'threshold'):
+    for algo in ('lasso_cd', 'threshold'):
         code = sparse_encode(X, V, algorithm=algo, positive=positive)
         if positive:
             assert (code >= 0).all()
         else:
             assert (code < 0).any()
 
-    try:
-        sparse_encode(X, V, algorithm='omp', positive=positive)
-    except ValueError:
-        if not positive:
-            raise
+    for algo in ('lasso_lars', 'lars', 'omp'):
```

Comment:
I would split the test into 2 tests. One that should raise error for not supporting the parameter and another checking that the coefficient are positive